### PR TITLE
Add Subscriptions List screen for V4 APIs

### PR DESCRIPTION
**Pull request by phiz71**
## Issue

https://gravitee.atlassian.net/browse/APIM-1519
https://gravitee.atlassian.net/browse/APIM-1987

## Description

First PR to display the list of Subscriptions.

What has been done:
 - 	creation of a new component with same layout as legacy one
 - bind query param in the console URL and the fields in the form
 - list subscriptions according to filters
 - search is done every time a field changes in the filters

TODO in other PRs:
 - fill the list of plans to allow filtering
 - fill the list of applications to allow filtering
 - export as CSV ? (Maybe not mandatory for v4.0.0)
 - create a subscription
 - plug the subscription detail screen (https://gravitee.atlassian.net/browse/APIM-1405)
 - clear filters button
 - search button vs. search on each form modification?

image

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wpklgxajvg.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/entities/management-api-v2/apiSubscriptionsResponse.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Gravitee.io APIM - Management API
+ * This is the OpenAPI specification for our new version of APIM Management API.
+ *
+ * OpenAPI spec version: 2.0.0
+ * Contact: team-apim@graviteesource.com
+ *
+ * NOTE: This class is auto generated by the swagger code generator program.
+ * https://github.com/swagger-api/swagger-codegen.git
+ * Do not edit the class manually.
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/entities/management-api-v2/apiSubscriptionsResponse.ts]: maybe we can remove this 😁

Comment (phiz71) [reply gravitee-apim-console-webui/src/entities/management-api-v2/apiSubscriptionsResponse.ts]: oups 😅

File: gravitee-apim-console-webui/src/management/api/api-ng-navigation/api-ng-navigation.component.ts [hunk 2]
```diff
@@ -80,6 +80,10 @@ export class ApiNgNavigationComponent implements OnInit {
 
     this.selectedItemWithTabs = this.findMenuItemWithTabs();
     this.breadcrumbItems = this.computeBreadcrumbItems();
+
+    this.ajsRootScope.$on('$locationChangeStart', () => {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/api-ng-navigation/api-ng-navigation.component.ts:84]: oups missing invest. i wanted to find another way to do that now we have a new routing :/

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      justify-content: flex-end;
+      align-items: center;
+
+      & button {
+        margin-left: 8px;
+      }
+    }
+
+    ::ng-deep .mat-form-field-wrapper {
+      margin-bottom: -1.25em;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss]: can we make it more specifique ? like  
```
.MyUniqueClass .mat-form-field-wrapper : {
```

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      width: 0;
+      cursor: move;
+    }
+
+    .mat-column-name {
+      width: 30%;
+    }
+  }
+
+  :where(.mat) .mat-form-field-wrapper {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss]: needed ?

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+              ['application', 'plan'],
+            )
+            .pipe(
+              catchError(() => {
+                this.snackBarService.error('Unable to try the request, please try again');
+                return EMPTY;
+              }),
+            ),
+        ),
+        catchError((error) => {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.ts]: maybe we can remove this one. if we prefer have the one before ?

Comment (phiz71) [reply gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.ts]: I lost myself in copy-paste and refactoring 😆

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss (hunk 4 above)
Comment (phiz71) [reply gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss]: No, it was a try I forgot to remove

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss (hunk 3 above)
Comment (phiz71) [reply gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.scss]: I've tried a lot of things but nothing else worked. We can have a look together if you want.

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.html [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+
+    <!-- Row shown when there is no data -->
+    <tr class="mat-row" *matNoDataRow>
+      <td *ngIf="!isLoadingData && subscriptionsTableDS.length === 0" class="mat-cell" [attr.colspan]="displayedColumns.length">
+        {{ 'There is no subscription (yet).' }}
+      </td>
+      <td *ngIf="isLoadingData" class="mat-cell" [attr.colspan]="displayedColumns.length">
+        {{ 'Loading...' }}
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.html:194]: We could use `gio-loader` here (like done here https://github.com/gravitee-io/gravitee-api-management/blob/30d82f2fd7c62d78beaa6f80b17aac8110b44dcd/gravitee-apim-console-webui/src/management/tasks/tasks.component.html#L51)

**Review by loriepisicchio — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.html (hunk 6 above)
Comment (phiz71) [reply gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.html:194]: Is it ok for you if do this in another PR?

**Review by ThibaudAV — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/api-ng-navigation/api-ng-navigation.component.ts (hunk 2 above)
Comment (phiz71) [reply gravitee-apim-console-webui/src/management/api/api-ng-navigation/api-ng-navigation.component.ts:84]: Will be taken in https://gravitee.atlassian.net/browse/APIM-1986

File: gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.html (hunk 6 above)
Comment (phiz71) [reply gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/list/api-portal-subscription-list.component.html:194]: https://gravitee.atlassian.net/browse/APIM-1990

**Pull request merged**